client/src/main/java/client/cs4224c/transaction/delivery/DeliveryTransaction.java

```diff
@@ -0,0 +1,66 @@
+package client.cs4224c.transaction.delivery;
+
+import client.cs4224c.transaction.AbstractTransaction;
+import client.cs4224c.transaction.delivery.data.DeliveryTransactionData;
+import client.cs4224c.util.PStatement;
+import client.cs4224c.util.QueryExecutor;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.List;
+
+public class DeliveryTransaction extends AbstractTransaction{
+
+    private final Logger logger = LoggerFactory.getLogger(DeliveryTransaction.class);
+
+    private DeliveryTransactionData data;
+
+    public DeliveryTransactionData getData() {
+        return data;
+    }
+
+    public void setData(DeliveryTransactionData data) {
+        this.data = data;
+    }
+
+    private static int INDEX_MIN_UNDELIVERED_ORDER_ID = 0;
+
+    private static int INDEX_CUSTOMER_ID = 0;
+
+    private static int INDEX_ORDER_LINE_NUMBER = 0;
+    private static int INDEX_ORDER_LINE_AMOUNT = 1;
+
+    private static int INDEX_CUSTOMER_BALANCE = 0;
+    private static int INDEX_CUSTOMER_DELIVERY_COUNT = 1;
+
+    @Override
+    public void executeFlow() {
```

Comment:
Give tests cases to ensure the logic is correct. I will tell you how to write today.